src/include/catalog/catalog.h

```diff
@@ -1,5 +1,6 @@
 #pragma once
 
+#include "catalog/catalog_entry/function_catalog_entry.h"
```

Comment:
forward declare `FunctionCatalogEntry` otherwise this will be propagated through includes.